### ISSUE TITLE
Odd Out of Bounds exception on world save - 1.7.10

### ISSUE THREAD

**Issue by Bysokar** [BuildCraft/BuildCraft#3611]
Hi Guys,

Hope you still support 1.7.10!!

Saw this with your latest version pop up in the server logs, not sure what the problem was exactly though.
Using v7.1.21.

```
[05:38:03 WARN]: Exception in thread "Thread-58"
[05:38:03 WARN]: java.lang.ArrayIndexOutOfBoundsException: 7005
[05:38:03 WARN]:        at gnu.trove.set.hash.TLongHashSet.toArray(TLongHashSet.java:167)
[05:38:03 WARN]:        at buildcraft.robotics.map.MapWorld.save(MapWorld.java:88)
[05:38:03 WARN]:        at buildcraft.robotics.map.MapManager.saveAllWorlds(MapManager.java:131)
[05:38:03 WARN]:        at buildcraft.robotics.map.MapManager.run(MapManager.java:144)
[05:38:03 WARN]:        at java.lang.Thread.run(Thread.java:745)
```

**Comment by AlexIIL** [comment]
Looks like we don't synchronise here https://github.com/BuildCraft/BuildCraft/blob/7.1.x/common/buildcraft/robotics/map/MapWorld.java#L143. As this is probably a concurrency issue there's probably not a good way of reproducing this to test.

**Comment by asiekierka** [comment]
I think throwing a synchronization in there would help, though.

**Comment by AlexIIL** [comment]
Should be fixed in [7.1.22](http://www.mod-buildcraft.com/buildcraft-7994-alpha-7122-stable.html)

**Comment by Bysokar** [comment]
Thanks guys - really appreciate it :)
